backend/pkg/httpserver/cache.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpserver
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log/slog"
+
+	"github.com/GoogleChrome/webstatus.dev/lib/cachetypes"
+)
+
+// operationResponseCache caches operation results using a RawBytesDataCacher.
+// It uses the `operationID` as a prefix for all cache keys, enabling
+// logical grouping and potential future deletion by prefix.  It also handles
+// JSON serialization/deserialization of keys and values.
+type operationResponseCache[Key any, Response any] struct {
+	cacher      RawBytesDataCacher
+	operationID string
+}
+
+func (c operationResponseCache[Key, Response]) key(key []byte) string {
+	return c.operationID + "-" + string(key)
```

Comment:
Will this cache persist across deployments?  Or does each deployed version of the app start with an empty cache?

Reply:
Good question!

Each version has its own copy. When we create the valkey cache, it reads in the K_REVISION from the environment variables:

https://github.com/GoogleChrome/webstatus.dev/blob/fd26e515654c54e9418f8ea9579ce6b2682a0fd9/backend/cmd/server/main.go#L81-L89

So there's already a prefix https://github.com/GoogleChrome/webstatus.dev/blob/fd26e515654c54e9418f8ea9579ce6b2682a0fd9/lib/valkeycache/cache.go#L60-L63

And [K_REVISION](https://cloud.google.com/run/docs/container-contract#:~:text=hello%2Dworld-,K_REVISION,-The%20name%20of) is provided by cloud run in GCP.